ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,172 @@
+    {
+        NODE_VALIDATION_CHECK(this,
+                              seq_len_pshape.rank().get_length() == 1,
+                              "The rank of sequence len tensor must be equal to 1.");
+    }
+
+    // check optional input type: blank index
+    if (get_input_size() == 3)
+    {
+        const auto& blank_index_type = get_input_element_type(2);
+        NODE_VALIDATION_CHECK(this,
+                              blank_index_type.is_integral_number(),
+                              "The blank index type is expected to be an integer type. Got: ",
+                              blank_index_type);
+
+        const auto& blank_index_shape = get_input_partial_shape(2);
+        Shape blank_index_type_shape = blank_index_shape.to_shape();
+        NODE_VALIDATION_CHECK(
+            this,
+            blank_index_shape.is_dynamic() || ngraph::is_scalar(blank_index_type_shape) ||
+                (is_vector(blank_index_type_shape) && (blank_index_type_shape[0] == 1)),
+            "Expected 0D or 1D tensor for the 'blank_index' input. Got: ",
+            blank_index_type);
+    }
+
+    // validate input shapes and compute output shape
+    ngraph::Dimension batch_size = Dimension::dynamic();
+    ngraph::Dimension time_size = Dimension::dynamic();
+
+    if (logits_pshape.rank().is_static())
+    {
+        if (logits_pshape[0].is_static())
+        {
+            batch_size = logits_pshape[0];
+        }
+        if (logits_pshape[1].is_static())
+        {
+            time_size = logits_pshape[1];
+        }
+    }
+
+    if (seq_len_pshape.rank().is_static())
+    {
+        if (seq_len_pshape[0].is_static())
+        {
+            if (batch_size != Dimension::dynamic())
+            {
+                NODE_VALIDATION_CHECK(this,
+                                      seq_len_pshape[0] == batch_size,
+                                      "The first dimensions of input tensors must match.");
+            }
+            batch_size = seq_len_pshape[0];
+        }
+    }
+
+    if (logits_pshape.rank().is_static() && seq_len_pshape.rank().is_static())
+    {
+        if (logits_pshape[0] == Dimension::dynamic() && seq_len_pshape[0] == Dimension::dynamic())
+        {
+            batch_size = seq_len_pshape[0] & logits_pshape[0];
+        }
+    }
+
+    set_output_type(0, m_classes_index_type, PartialShape{batch_size, time_size});
+    set_output_type(1, m_sequence_length_type, PartialShape{batch_size});
+}
+
+bool op::v6::CTCGreedyDecoderSeqLen::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("merge_repeated", m_merge_repeated);
+    visitor.on_attribute("classes_index_type", m_classes_index_type);
+    visitor.on_attribute("sequence_length_type", m_sequence_length_type);
+    return true;
+}
+
+shared_ptr<Node>
+    op::v6::CTCGreedyDecoderSeqLen::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+
+    if (new_args.size() == 2)
```

Comment:
const method new_args.size() is calling twice

Reply:
Not sure that it need optimize.
Size calculated before calling method.